torch/_dynamo/variables/optimizer.py

```diff
@@ -62,7 +62,8 @@ def call_method(
                 self.update_list_args(tx, args, kwargs, py_args, py_kwargs)
                 # stash a weak_ptr to optimizer to invalidate code
                 # if the optimizer object dies
-                tx.store_global_weakref(self.get_global_name(), self.value)
+                mangled_name = f"__optimizer_{id(self.value)}"
+                tx.store_global_weakref(mangled_name, self.value)
```

Comment:
I don't quite understand what the rule for the refactor was here

Reply:
This is the only use of self.get_global_name(), so I inlined it here.

Reply:
Not applicable anymore